src/Messages/OWSMessageSender.m

```diff
@@ -486,36 +487,7 @@ - (void)resendMessageFromKeyError:(TSInvalidIdentityKeySendingErrorMessage *)err
     //  2.) fail and create a new identical error message in the thread.
     [errorMessage remove];
 
-    if ([errorMessage.thread isKindOfClass:[TSContactThread class]]) {
```

Comment:
That's interesting. I'm inclined to leave the new state tracking which recipients have received the message.  I'm hoping it improves the experience when sending messages to very large groups (100 or 1,000 members) from bad connections, where the odds of sending failures is high - by tracking "sent recipients", we can "resume" the send from where the error took place.  WDYT?